### ISSUE TITLE
http: `undefined` headers should not be sent

### ISSUE THREAD

**Issue by bminer** [joyent/node#7305]
If an HTTP header value is `undefined`, that particular header should not be transmitted to the server.

Example: 

```
var http = require("http");
http.request({
    "host": "google.com",
    "headers": {"If-None-Match": undefined}
});
```

In the case above, the `If-None-Match` header is set to the string `"undefined"`.  Instead, this header should not be sent at all.

The string conversion happens here... see https://github.com/joyent/node/blob/2eddd7411234bf633df6a7bdb9c120ed89853324/lib/_http_outgoing.js#L301

**Comment by jasnell** [comment]
Appears to have been resolved in v0.12, but still happens in v0.10. @joyent/node-coreteam ... should we do a backport or just leave it?

**Comment by trevnorris** [comment]
@jasnell if the fix is trivial then I'd say backport.

**Comment by jasnell** [comment]
At this point, with v0.10 in maintenance, backporting is likely unnecessary. It's fixed in v0.12 and v3.x.
